### PR TITLE
#20 [Stablish a general layout for the app]

### PR DESCRIPTION
**Pull request by Efgar**
Story: https://github.com/TFFLabs/SoundFoundry/issues/20

This adds some layout and some minor refactor over authentication flow.

### CODE REVIEW

File: src/app/services/user.service.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+export class UserService {
+  devices: Devices = new Devices();
+  user: User = new User();
+
+  constructor(private spotifyService: SpotifyService) {
+    this.getUser();
+    this.getUserDevices();
+  }
+
+  public getUser() {
```
Comment (dahernandezt) [inline src/app/services/user.service.ts]: I think it applies the same comment on the get methods...we can change it to  "load" maybe... also.. is this method used outside the class?  If it is called only in the init method,  can we change them to "private"? (I am not sure, maybe we can verify with Jorge).

File: src/app/services/user.service.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    Promise.resolve(
+      this.spotifyService.getUser().subscribe(user => {
+        this.user = new User().deserialize(user);
+      })
+    );
+  }
+
+  public getUserDevices() {
```
Comment (dahernandezt) [inline src/app/services/user.service.ts]: See comment Above.

File: src/app/services/authorization.service.ts [hunk 3]
```diff
@@ -46,7 +47,7 @@ export class AuthorizationService {
           console.log('Token: '+ e.newValue);
           this.authToken.value = e.newValue;
           window.removeEventListener('storage', storageChanged, false);
-
+          this.spotifyService.token = e.newValue;
```
Comment (dahernandezt) [inline src/app/services/authorization.service.ts:50]: We are saving this token value in multiple variables/classes, are we using all of them? maybe we need to unifiy them in only one place. (See line 48... this.authToken.value = e.newValue;)

**Review by dahernandezt — APPROVED** [review]

**Review by dahernandezt — CHANGES_REQUESTED** [review]
See comments on code

**Review by dahernandezt — APPROVED** [review]

File: src/app/services/authorization.service.ts (hunk 3 above)
Comment (Efgar) [reply src/app/services/authorization.service.ts:50]: This shall be handled in #19 and #14, we do need to figure the best way to save the token.

**Pull request merged**